src/Compilers/CSharp/Test/CommandLine/CommandLineDiagnosticFormatterTests.cs

```diff
@@ -28,5 +28,39 @@ public void GetPathNameRelativeToBaseDirectory()
             Assert.Equal(@"temp\a.cs", formatter.RelativizeNormalizedPath(@"X:\rootdir\dir\temp\a.cs"));
             Assert.Equal(@"Y:\rootdir\dir\a.cs", formatter.RelativizeNormalizedPath(@"Y:\rootdir\dir\a.cs"));
         }
+
+        [ConditionalFact(typeof(WindowsOnly))]
+        [WorkItem(14725, "https://github.com/dotnet/roslyn/issues/14725")]
+        public void RelativizeNormalizedPathShouldHandleRootPaths()
+        {
+            var formatter = new CommandLineDiagnosticFormatter(
+                baseDirectory: @"C:\",
+                displayFullPaths: false,
+                displayEndLocations: false);
+
+            Assert.Equal(@"temp.cs", formatter.RelativizeNormalizedPath(@"c:\temp.cs"));
+        }
+
+        [ConditionalFact(typeof(WindowsOnly))]
+        public void RelativizeNormalizedPathShouldHandleDirectoriesWithSamePrefix_1()
+        {
+            var formatter = new CommandLineDiagnosticFormatter(
+                baseDirectory: @"C:\AB",
+                displayFullPaths: false,
+                displayEndLocations: false);
+
+            Assert.Equal(@"c:\ABCD\file.cs", formatter.RelativizeNormalizedPath(@"c:\ABCD\file.cs"));
+        }
+
+        [ConditionalFact(typeof(WindowsOnly))]
+        public void RelativizeNormalizedPathShouldHandleDirectoriesWithSamePrefix_2()
+        {
+            var formatter = new CommandLineDiagnosticFormatter(
+                baseDirectory: @"C:\ABCD",
+                displayFullPaths: false,
+                displayEndLocations: false);
+
+            Assert.Equal(@"c:\AB\file.cs", formatter.RelativizeNormalizedPath(@"c:\AB\file.cs"));
+        }
```

Comment:
Please also test sub-directories between `baseDirectory` and `normalizedPath`. e.g.: `baseDirectory: "C:\A\B"`, `normalizedPath: "C:\A\B\C\D\file.cs"`